### PR TITLE
refact: unify naming of degree for oltp algorithms

### PR DESCRIPTION
**Pull request by imbajin**
Note: also keep `degree` for alias in `Step*` request

### CODE REVIEW

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/CountAPI.java [hunk 1]
```diff
@@ -123,22 +123,22 @@ public String toString() {
         public List<String> labels;
         @JsonProperty("properties")
         public Map<String, Object> properties;
-        @JsonProperty("degree")
-        public long degree = Long.valueOf(DEFAULT_DEGREE);
+        @JsonProperty("max_degree")
+        public long maxDegree = Long.valueOf(DEFAULT_MAX_DEGREE);
```
Comment (Linary) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/CountAPI.java:127]: replace Long.valueOf() -> Long.parseLong, remove redundant boxing

**Review by Linary — CHANGES_REQUESTED** [review]

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/CountAPI.java (hunk 1 above)
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/CountAPI.java:127]: not addressed Linary's comment

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/CustomizedCrosspointsAPI.java [hunk 2]
```diff
@@ -173,24 +173,24 @@ public String toString() {
         public List<String> labels;
         @JsonProperty("properties")
         public Map<String, Object> properties;
-        @JsonProperty("degree")
-        public long degree = Long.valueOf(DEFAULT_DEGREE);
+        @JsonProperty("max_degree")
+        public long maxDegree = Long.valueOf(DEFAULT_MAX_DEGREE);
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/CustomizedCrosspointsAPI.java:177]: ditto

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/FusiformSimilarityAPI.java [hunk 3]
```diff
@@ -75,9 +75,9 @@ public String post(@Context GraphManager manager,
         E.checkArgument(request.minNeighbors > 0,
                         "The min neighbor count must be > 0, but got: %s",
                         request.minNeighbors);
-        E.checkArgument(request.degree > 0 || request.degree == NO_LIMIT,
-                        "The degree of request must be > 0, but got: %s",
-                        request.degree);
+        E.checkArgument(request.maxDegree > 0 || request.maxDegree == NO_LIMIT,
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/FusiformSimilarityAPI.java:78]: 0L

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/PersonalRankAPI.java [hunk 4]
```diff
@@ -68,9 +68,9 @@ public String personalRank(@Context GraphManager manager,
         E.checkArgument(request.alpha > 0 && request.alpha <= 1.0,
                         "The alpha of rank request must be in range (0, 1], " +
                         "but got '%s'", request.alpha);
-        E.checkArgument(request.degree > 0 || request.degree == NO_LIMIT,
-                        "The degree of rank request must be > 0, but got: %s",
-                        request.degree);
+        E.checkArgument(request.maxDegree > 0 || request.maxDegree == NO_LIMIT,
+                        "The maxDegree of rank request must be > 0, but got: %s",
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/PersonalRankAPI.java:72]: The max_degree of rank request.
same as other check

File: hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/steps/WeightedEdgeStep.java [hunk 5]
```diff
@@ -59,24 +59,24 @@ public WeightedEdgeStep(HugeGraph g, Directions direction, List<String> labels)
...
                             String weightBy, double defaultWeight, long sample) {
         E.checkArgument(sample > 0L || sample == NO_LIMIT,
                         "The sample must be > 0 or == -1, but got: %s",
                         sample);
-        E.checkArgument(degree == NO_LIMIT || degree >= sample,
-                        "Degree must be greater than or equal to sample," +
-                        " but got degree %s and sample %s",
-                        degree, sample);
+        E.checkArgument(maxDegree == NO_LIMIT || maxDegree >= sample,
+                        "Max degree must be greater than or equal to sample," +
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/steps/WeightedEdgeStep.java:74]: The max_degree must...

File: hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/steps/WeightedEdgeStep.java [hunk 6]
```diff
@@ -59,24 +59,24 @@ public WeightedEdgeStep(HugeGraph g, Directions direction, List<String> labels)
...
+                        " but got max degree %s and sample %s",
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/steps/WeightedEdgeStep.java:75]: max_degree

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/PersonalRankAPI.java [hunk 7]
```diff
@@ -68,14 +68,14 @@ public String personalRank(@Context GraphManager manager,
         E.checkArgument(request.alpha > 0 && request.alpha <= 1.0,
                         "The alpha of rank request must be in range (0, 1], " +
                         "but got '%s'", request.alpha);
-        E.checkArgument(request.maxDegree > 0 || request.maxDegree == NO_LIMIT,
-                        "The maxDegree of rank request must be > 0, but got: %s",
-                        request.maxDegree);
-        E.checkArgument(request.limit > 0 || request.limit == NO_LIMIT,
+        E.checkArgument(request.maxDegree > 0L || request.maxDegree == NO_LIMIT,
+                        "The max_degree of rank request must be > 0," +
+                        " but got: %s", request.maxDegree);
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/PersonalRankAPI.java:73]: prefer to put space to the previous line

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/NeighborRankAPI.java [hunk 8]
```diff
@@ -123,8 +123,8 @@ public String toString() {
         public Directions direction;
         @JsonProperty("labels")
         public List<String> labels;
-        @JsonProperty("degree")
-        public long degree = Long.parseLong(DEFAULT_DEGREE);
+        @JsonProperty("max_degree")
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/NeighborRankAPI.java:126]: keep compatible through alias "degree"

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/TraverserAPI.java [hunk 9]
```diff
@@ -45,17 +45,17 @@ protected static EdgeStep step(HugeGraph graph, Step step) {
         public List<String> labels;
         @JsonProperty("properties")
         public Map<String, Object> properties;
-        @JsonProperty("degree")
-        public long degree = Long.valueOf(DEFAULT_DEGREE);
+        @JsonProperty("max_degree")
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/TraverserAPI.java:48]: ditto

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/PersonalRankAPI.java [hunk 10]
```diff
@@ -106,10 +106,10 @@ public String personalRank(@Context GraphManager manager,
         private String label;
         @JsonProperty("alpha")
         private double alpha;
-        @JsonProperty("degree")
-        private long degree = Long.valueOf(DEFAULT_DEGREE);
+        @JsonProperty("max_degree")
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/PersonalRankAPI.java:109]: this param is not in a step, don't need alias

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/CustomizedPathsAPI.java [hunk 11]
```diff
@@ -160,31 +160,31 @@ public String toString() {
         public List<String> labels;
         @JsonProperty("properties")
         public Map<String, Object> properties;
-        @JsonProperty("degree")
-        public long degree = Long.valueOf(DEFAULT_DEGREE);
+        @JsonProperty("max_degree")
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/CustomizedPathsAPI.java:163]: ditto

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/CustomizedCrosspointsAPI.java [hunk 12]
```diff
@@ -173,24 +173,24 @@ public String toString() {
         public List<String> labels;
         @JsonProperty("properties")
         public Map<String, Object> properties;
-        @JsonProperty("degree")
-        public long degree = Long.valueOf(DEFAULT_DEGREE);
+        @JsonProperty("max_degree")
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/CustomizedCrosspointsAPI.java:176]: ditto

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/CountAPI.java [hunk 13]
```diff
@@ -123,22 +123,22 @@ public String toString() {
         public List<String> labels;
         @JsonProperty("properties")
         public Map<String, Object> properties;
-        @JsonProperty("degree")
-        public long degree = Long.valueOf(DEFAULT_DEGREE);
+        @JsonProperty("max_degree")
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/CountAPI.java:126]: keep compatible through alias "degree"
note import com.fasterxml.jackson.annotation.JsonAlias

File: hugegraph-api/src/main/java/com/baidu/hugegraph/version/ApiVersion.java [hunk 14]
```diff
@@ -61,7 +61,7 @@
      * [0.20] HugeGraph-1210: Add paths api to find paths between two nodes
      * [0.21] HugeGraph-1197: Expose scan api for hugegraph-spark
      * [0.22] HugeGraph-1162: Support authentication and permission control
-     * [0.23] HugeGraph-1176: Support degree and capacity for traverse api
+     * [0.23] HugeGraph-1176: Support maxDegree and capacity for traverse api
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/version/ApiVersion.java:64]: don't update it

File: hugegraph-api/src/main/java/com/baidu/hugegraph/version/ApiVersion.java [hunk 15]
```diff
@@ -75,7 +75,7 @@
      * [0.31] Issue-182: Support restore graph in restoring and merging mode
      *
      * version 0.9:
-     * [0.32] Issue-250: Keep depth and degree consistent for traverser api
+     * [0.32] Issue-250: Keep depth and maxDegree consistent for traverser api
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/version/ApiVersion.java:78]: ditto

File: hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/steps/WeightedEdgeStep.java [hunk 16]
```diff
@@ -59,24 +59,24 @@ public WeightedEdgeStep(HugeGraph g, Directions direction, List<String> labels)
     public WeightedEdgeStep(HugeGraph g, Directions direction, List<String> labels,
                             Map<String, Object> properties) {
         this(g, direction, labels, properties,
-             Long.valueOf(DEFAULT_DEGREE), 0L, null, 0.0D,
+             Long.valueOf(DEFAULT_MAX_DEGREE), 0L, null, 0.0D,
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/steps/WeightedEdgeStep.java:62]: also use parseLong?

**Review by javeme — APPROVED** [review]

**Review by zhoney — APPROVED** [review]

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/AllShortestPathsAPI.java [hunk 17]
```diff
@@ -22,7 +22,7 @@
 import java.util.List;
 
 import static com.baidu.hugegraph.traversal.algorithm.HugeTraverser.DEFAULT_CAPACITY;
-import static com.baidu.hugegraph.traversal.algorithm.HugeTraverser.DEFAULT_DEGREE;
+import static com.baidu.hugegraph.traversal.algorithm.HugeTraverser.DEFAULT_MAX_DEGREE;
```
Comment (zhoney) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/AllShortestPathsAPI.java:25]: prefer upgrade server api version

File: hugegraph-api/src/main/java/com/baidu/hugegraph/version/ApiVersion.java [hunk 18]
```diff
@@ -110,10 +110,11 @@
      *        multi-node-shortest-path, jaccard-similar and template-paths
      * [0.59] Issue-1333: Support graph read mode for olap property
      * [0.60] Issue-1392: Support create and resume snapshot
+     * [0.61] Issue-1433: Unify naming of degree for olap algorithms
```
Comment (zhoney) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/version/ApiVersion.java:113]: olap -> oltp

**Review by zhoney — CHANGES_REQUESTED** [review]

**Review by zhoney — APPROVED** [review]

**Review by javeme — APPROVED** [review]

**Review by Linary — APPROVED** [review]

**Pull request merged**